### PR TITLE
TechDocs Homepage

### PR DESCRIPTION
**Pull request by emmaindal**
Closes: #4103
Closes:  #4102 

Screen Shot 2021-02-05 at 11 31 25 AM
Screen Shot 2021-02-05 at 11 31 19 AM

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

File: plugins/techdocs/src/reader/components/OverviewContent.tsx
Comment (OrkoHunter) [inline plugins/techdocs/src/reader/components/OverviewContent.tsx:1]: I wonder if it makes sense to put this and other homepage components out of "reader/components/" directory. 🤔

File: plugins/techdocs/src/reader/components/OverviewContent.tsx
Comment (OrkoHunter) [inline plugins/techdocs/src/reader/components/OverviewContent.tsx:53]: `entity.metadata.description` could be `undefined`.
```suggestion
                  description={entity.metadata.description ?? ''}
```

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

**Review by OrkoHunter — COMMENTED** [review]
Awesome! Will run it locally later today to see how it looks. :)

File: plugins/techdocs/src/reader/components/OverviewContent.tsx
Comment (emmaindal) [reply plugins/techdocs/src/reader/components/OverviewContent.tsx:1]: yeah I was thinking about that actually. I'll take the opportunity to move out all components from reader in this PR as well!

Comment (OrkoHunter) [reply plugins/techdocs/src/reader/components/OverviewContent.tsx:1]: Nice!

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

File: plugins/techdocs/src/home/components/OwnedContent.tsx
Comment (dhenneke) [inline plugins/techdocs/src/home/components/OwnedContent.tsx:47]: Would be cool to also see documents that are owned via group membership. `plugin-catalog-react` has a [`isOwnerOf`](https://github.com/backstage/backstage/blob/2cc2d1ad1788d5680d77b6192e40870d14a83f0c/plugins/catalog-react/src/utils/isOwnerOf.ts#L29) helper for this which is also used in the catalog. However, the code to find the user entity is is not directly usable here and also not very complete since it waits for the new user-identity mapping ([`useOwnUser`](https://github.com/backstage/backstage/blob/2cc2d1ad1788d5680d77b6192e40870d14a83f0c/plugins/catalog/src/components/useOwnUser.ts#L26)).

Comment (emmaindal) [reply plugins/techdocs/src/home/components/OwnedContent.tsx:47]: @dhenneke Very good point. Thanks for pointing that out. I'll see what the status is of `useOwnUser` otherwise I might be able to use some workaround here until its ready!

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

File: plugins/techdocs/src/home/components/OwnedContent.tsx
Comment (andrewthauer) [inline plugins/techdocs/src/home/components/OwnedContent.tsx:109]: I think the search might be useful to enable here?

File: plugins/techdocs/src/home/components/OwnedContent.tsx
Comment (andrewthauer) [inline plugins/techdocs/src/home/components/OwnedContent.tsx:53]: Not directly related to this but just a general idea / thought.  Now that systems and domains are supported in the catalog, maybe it would be worth opening up docs support to more then just components. I suppose the difference being the actual docs site would be a component, but it could be associated with a different entity kind via an annotations maybe. Anyways not in scope for this PR, just thinking.

**Review by andrewthauer — COMMENTED** [review]
Nice to see these changes 😀. I'm wondering if the overview page might get a little big for the cards view though. I can see it being nice for a curated list of docs sites, but perhaps harder to search for all docs sites. Perhaps there are additional stories around this?

File: plugins/techdocs/src/home/components/OwnedContent.tsx
Comment (emmaindal) [reply plugins/techdocs/src/home/components/OwnedContent.tsx:109]: Will combine my answer to this with the general feedback above. Haven't spent any time in changing the "Overview" page right now (except from adding a title) because yes, it will include a full search experience when search is ready. You can see more designs of that in this [issue](https://github.com/backstage/backstage/issues/2028). So we just left that out for now and focused on the owned page. With that said, i'll enable search on the table for now as it makes sense until we have a full search page. Thanks for pointing that out!

File: plugins/techdocs/src/home/components/OwnedContent.tsx
Comment (emmaindal) [reply plugins/techdocs/src/home/components/OwnedContent.tsx:53]: Hmm yes agree, great idea, I was just thinking "do we have any restrictions on that `kind` needs to be Component for TechDocs"? I guess we do support more then that 🤔  But I might be wrong, @OrkoHunter maybe can give us the latest update here!

Comment (OrkoHunter) [reply plugins/techdocs/src/home/components/OwnedContent.tsx:53]: Hmm, I don't think there is any restriction to `kind` being `Component` for TechDocs and it could be anything, as long as there is `backstage.io/techdocs-ref` annotation in the entity.

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

File: plugins/techdocs/src/home/components/OwnedContent.tsx
Comment (andrewthauer) [reply plugins/techdocs/src/home/components/OwnedContent.tsx:53]: Haven't kept up with the development, but is there a possibility of conflicting behaviour if the same techdocs annotation is used in multiple entities. For example it's attached to the component of type documentation for the repo that provides the docs and a domain or system that they are related to.

Comment (OrkoHunter) [reply plugins/techdocs/src/home/components/OwnedContent.tsx:53]: There shouldn't be any conflicts if two Entities have same annotation. The generated sites are stored in and retried from `:namespace/:kind/:name/` directory in the storage. So, same techdocs-ref annotation for multiple entities only means that the source code is common, but the sites are going to be unique, since both are different entities. Hope this makes sense.

Comment (andrewthauer) [reply plugins/techdocs/src/home/components/OwnedContent.tsx:53]: Perfect. That makes this idea already supported :)

Comment (emmaindal) [reply plugins/techdocs/src/home/components/OwnedContent.tsx:53]: @andrewthauer The thing I can think of now is of course that if you do have a system-kind entity pointing to a techdocs ref for documentation and then you have a domain-kind entity pointing to the same techdocs ref as you would like these entities to point to the same documentation. That will cause publishing of two different docs sites, where the documentation is the same, right? @OrkoHunter So that one could see as kind of a conflict...

Comment (OrkoHunter) [reply plugins/techdocs/src/home/components/OwnedContent.tsx:53]: Yeah, sites will still be duplicated, but I think it's best that they are unique per entity. :)

Comment (andrewthauer) [reply plugins/techdocs/src/home/components/OwnedContent.tsx:53]: So we have this scenario now. We have domains and we also some domains have a dedicated docs-as-code repo for documentation related to the domain. Currently the docs only repo is registered as a component but it would be nice to natively associate it with its applicable domain. Are you saying we should register the repo with the domain entity instead of the component?

Comment (OrkoHunter) [reply plugins/techdocs/src/home/components/OwnedContent.tsx:53]: > Currently the docs only repo is registered as a component
> Are you saying we should register the repo with the domain entity instead of the component?

Yes! You shouldn't have to create a new Component just to view docs of a Domain.

Assuming you are on the latest techdocs and techdocs-backend plugins and using techdocs-cli on CI/CD ->

Publish the docs on CI with the CLI option `--entity` to be `:namespace/:kind/:name` such that `:kind` is `System` https://github.com/backstage/techdocs-cli#publish-generated-techdocs-sites

Add `backstage.io/techdocs-ref: url:TARGET` annotation to your domain's `catalog-info.yaml` to enable TechDocs site for it in Backstage frontend. And when you open the TechDocs site for it, the site will be fetched from the storage for the domain using the unique `:namespace/:kind/:name` in the URL of the site.

I should probably investigate more on Systems and Domains https://backstage.io/docs/features/software-catalog/system-model since they represent a collection of Entities but for TechDocs, they are not that different from an entity like Component as long as the triplet `:namespace/:kind/:name` is unique and entity can have a `backstage.io/techdocs-ref` metadata annotation.

File: plugins/techdocs/src/home/components/OverviewContent.tsx
Comment (egnwd) [inline plugins/techdocs/src/home/components/OverviewContent.tsx:39]: I was reading #4102 and it says 
> a Support Button should exist with the possibility for a Backstage integrator to insert information to the team responsible for TechDocs within the organisation

The support button here looks static without the possibility to change things like the `slackChannel`. I think it's inline with some other support buttons from other plugins, however it would be great if the support information could be injected into the component.

I point this out as while implementing backstage at my organisation I found that to modify things like the support button I was having to recreate whole pages and we miss out on great improvements like this owned docs tab. Unless of course I am missing a way to do this/this is the intended pattern.

Would like to know your thoughts, if this is something that could be included perhaps in this PR, or I can raise an issue if I've misunderstood #4102

Comment (OrkoHunter) [reply plugins/techdocs/src/home/components/OverviewContent.tsx:39]: We need to figure out a setup to let Backstage integrators configure "Support" information, not just for TechDocs but for all the plugins. Doing it via `app-config.yaml` is likely to be discouraged since the file is going to get inflated - this is likely to be done by code when a Backstage app is setup.

cc @backstage/maintainers

Comment (OrkoHunter) [reply plugins/techdocs/src/home/components/OverviewContent.tsx:39]: Related https://github.com/backstage/backstage/pull/4517

`app.support` in `app-config.yaml` might be a good start.

**Comment by OrkoHunter** [comment]
We'll wait until the `app.support` feature is merged in #4517 and reuse it for Support details on TechDocs homepage.

**Comment by freben** [comment]
Hey - I noticed now that we've been skimming past this one but it actually needs approval. Sorry if that's been blocking you. Is it ready on your end? Apart from the merge conflict.

**Comment by freben** [comment]
Oh sorry, now I read the support comment.

**Comment by emmaindal** [comment]
yeah no worries @freben its a little blocked right now and we will continue on it soon! Thank you anyways 👏

**Comment by adamdmharvey** [comment]
FYI should be unblocked now - `app.support` is in `master`.

**Comment by andrewthauer** [comment]
> FYI should be unblocked now - `app.support` is in `master`.

Oh, sorry. I didn't even realize this was blocked on that. Hopefully it works for the purposes of this PR.

**Comment by emmaindal** [comment]
Nice, thank you for the update @adamdmharvey, I'll look into it tomorrow and hopefully get it 🚢

**Comment by andrewthauer** [comment]
So just a heads up. The updated SupportButton hasn't been fully extended for plugins. You can insert child content and it does internal read from a new `app.support` key which supports multiple item groupings.

The idea was to extend this further to allow plugins to add their own support item groups or somehow do this automatically via reading some metadata from the package.json.

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

**Comment by emmaindal** [comment]
Understand @andrewthauer I just added some example support information around TechDocs for now. 
Screen Shot 2021-03-04 at 11 56 52 AM

But agree with that it would be nice to just be able to show TechDocs support config and not all configs on specific plugin pages.

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

File: plugins/techdocs/src/home/components/OwnedContent.tsx
Comment (emmaindal) [reply plugins/techdocs/src/home/components/OwnedContent.tsx]: > Would be cool to also see documents that are owned via group membership. `plugin-catalog-react` has a [`isOwnerOf`](https://github.com/backstage/backstage/blob/2cc2d1ad1788d5680d77b6192e40870d14a83f0c/plugins/catalog-react/src/utils/isOwnerOf.ts#L29) helper for this which is also used in the catalog. However, the code to find the user entity is is not directly usable here and also not very complete since it waits for the new user-identity mapping ([`useOwnUser`](https://github.com/backstage/backstage/blob/2cc2d1ad1788d5680d77b6192e40870d14a83f0c/plugins/catalog/src/components/useOwnUser.ts#L26)).

Updated to use this but @backstage/maintainers probably want to take a look if its ok to use or if we should wait. I exported the hook `useOwnUser` from the catalog plugin to be able to use it, let me know if I should change back again. 
Screen Shot 2021-03-04 at 1 11 37 PM

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

**Review by OrkoHunter — APPROVED** [review]

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

**Comment by emmaindal** [comment]
> Hey - I noticed now that we've been skimming past this one but it actually needs approval. Sorry if that's been blocking you. Is it ready on your end? Apart from the merge conflict.

nooow its ready for review from your side @freben

File: app-config.yaml
Comment (adamdmharvey) [inline app-config.yaml:12]: So the only issue with this I believe is this is global - this isn't JUST for TechDocs, it'll now add this link across the entire Backstage instance for every single support button on all components and all plugins.  Was that what was intended?

I have a feeling this should instead be done when/if we can extend the support component to inherit component- or plugin-specific support elements?

Comment (emmaindal) [reply app-config.yaml:12]: Yeah add it globally is the only option right now, so either that or leave it out of this PR.

Comment (emmaindal) [reply app-config.yaml]: @adamdmharvey I'll leave the support config for TechDocs out of this PR fyi!

File: packages/core-api/src/icons/icons.tsx
Comment (Rugvip) [inline packages/core-api/src/icons/icons.tsx:64]: This addition needs a changeset

File: plugins/catalog/src/index.ts
Comment (Rugvip) [inline plugins/catalog/src/index.ts:34]: Don't think we want this tbh, should probably be moved to the `catalog-react` package? Tbh it's not a huge thing either so could perhaps copy it instead

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

File: plugins/catalog/src/index.ts
Comment (emmaindal) [reply plugins/catalog/src/index.ts:34]: 👍 will do!

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

File: plugins/techdocs/package.json
Comment (Rugvip) [inline plugins/techdocs/package.json:38]: Should no longer be needed now right?

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4401-pvley4xwc81k5tq8nyd1ohuvajk3f1jm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601d20a098552d1fa65756d7

**Review by Rugvip — APPROVED** [review]

**Pull request merged**
